gguf-py/gguf/gguf_writer.py

```diff
@@ -293,43 +358,60 @@ def write_tensors_to_file(self, *, progress: bool = False) -> None:
 
         assert self.fout is not None
 
-        self.write_padding(self.fout, self.fout.tell())
+        for fout in self.fout:
+            self.write_padding(fout, fout.tell())
 
         if self.temp_file is None:
             bar = None
+            shard_bar = None
 
             if progress:
                 from tqdm import tqdm
 
-                total_bytes = sum(t.nbytes for t in self.tensors.values())
+                total_bytes = sum(ti.nbytes for t in self.tensors for ti in t.values())
 
                 bar = tqdm(desc="Writing", total=total_bytes, unit="byte", unit_scale=True)
-
-            # relying on the fact that Python dicts preserve insertion order (since 3.7)
-            for ti in self.tensors.values():
-                assert ti.tensor is not None  # can only iterate once over the tensors
-                assert ti.tensor.nbytes == ti.nbytes
-                ti.tensor.tofile(self.fout)
-                if bar is not None:
-                    bar.update(ti.nbytes)
-                self.write_padding(self.fout, ti.nbytes)
-                ti.tensor = None
+                if len(self.fout) > 1:
+                    shard_bar = tqdm(desc=f"Shard (1/{len(self.fout)})", total=total_bytes, unit="byte", unit_scale=True)
```

Comment:
I don't know about you, but from trying it a few times, I feel like the global progress `bar` should be on the bottom, while the `shard_bar` should be right above it. This simply requires swapping their order here (and also maybe in the other places for consistency).

This is subjective, but I like to have the slower-moving `bar` anchored on the bottom, and not sandwiched between the previous logs and a fast-moving `shard_bar`.

Reply:
We can do that. Personally I prefer it the other way but I see how you'd want the things that complete first on the top.